uqcsbot/utils/uq_course_utils.py

```diff
@@ -16,12 +16,87 @@
 OFFERING_PARAMETER = "offer"
 
 
+class Offering:
+    """
+    A semester, campus and mode (e.g. Internal) that many courses occur within
+    """
+
+    CampusType = Literal["St Lucia", "Gatton", "Herston"]
+    # The codes used internally within UQ systems
+    campus_codes: Dict[CampusType, str] = {
+        "St Lucia": "STLUC",
+        "Gatton": "GATTN",
+        "Herston": "HERST",
+    }
+
+    ModeType = Literal["Internal", "External", "Flexible Delivery"]
```

Comment:
Also Intensive mode for e.g. [MGTS7801](https://course-profiles.uq.edu.au/student_section_loader/section_1/120684), which is available under an ITEE-administered degree (MCyberSec).

I don't believe there are any others (maybe some sort of field-work/practical mode for medical students?) but this is something to be aware of if more modes crop up.